### DISCUSSION TITLE
about renegotiation

### DISCUSSION THREAD

**Discussion by junnylc** [wqweto/VbAsyncSocket#26] [Q&A] [answered]
I use ctxwinsock for my HTTPS server program. When I use myssl to scan the server program, myssl reports that the server program allows renegotiation, which is a serious security vulnerability. How do I prohibit renegotiation? thank you
myssl report: Support client insecure renegotiation

**Comment by wqweto** [comment, marked as answer]
Fixed in commit 4b093b65332dd52697a32c87ca9134f111009d74

10x for the bug report!

Edit: Did some tests with a VbAsyncSocket server with a valid Let's Encrypt issued certificate
1. [Test by myssl.com](https://myssl.com/free.bgbiz.com?domain=free.bgbiz.com&status=success)
2. [Test by ssllabs.com](https://www.ssllabs.com/ssltest/analyze.html?d=free.bgbiz.com&latest)

Something is wrong with myssl.com test because it says "protocol error" for `IE 8-10 / Win 7` handshake simulation but I just tested accessing the https server from Win7 (IE8) and TLS version 1.2 was negotiated w/ ECDHE-RSA-AES128-SHA just ok.

**Reply by junnylc** [reply]
thank you very very very much!!success

**Reply by junnylc** [reply]
you are my idol!!!

**Reply by junnylc** [reply]
I'd founf the error too
>Something is wrong with myssl.com test because it says "protocol error" for IE 8-10 / Win 7 handshake simulation but I just tested accessing the https server from Win7 (IE8) and TLS version 1.2 was negotiated w/ ECDHE-RSA-AES128-SHA just ok.

**Comment by junnylc** [comment]
Thank you. I have modified the code and the effect is very good.



At 2022-06-22 17:05:02, "Vladimir Vissoultchev" ***@***.***> wrote:

Fixed in commit 4b093b6

10x for the bug report!
